double count for UpstreamResponseFailed when upstream response timeout

```
2021-01-14 18:11:32,645 [INFO] [admin store] [start service] start service Mosn Admin Server on [::]:34901
2021-01-14 18:11:48,658 [INFO] [2,c0a801041610619098656100194832] [proxy] [downstream] onResponseTimeout，host: 127.0.0.1:2046, time: 1m0s
goroutine 98 [running]:
runtime/debug.Stack(0x516bc80, 0xc000349470, 0x5432cf1)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
mosn.io/mosn/pkg/proxy.(*downStream).onUpstreamReset(0xc000224580, 0x5432cf1, 0x15)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:1228 +0x2c6
mosn.io/mosn/pkg/proxy.(*downStream).processError(0xc000224580, 0x1, 0xc000224580, 0x5698b01, 0xc000388540)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:1587 +0x43c
mosn.io/mosn/pkg/proxy.(*downStream).waitNotify(0xc000224580, 0x1, 0x1, 0x0, 0x0)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:1563 +0x8e
mosn.io/mosn/pkg/proxy.(*downStream).receive(0xc000224580, 0x5698ba0, 0xc000388540, 0xc000000001, 0xb, 0xc00007df90)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:598 +0xd04
mosn.io/mosn/pkg/proxy.(*downStream).OnReceive.func1()
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:406 +0xca
mosn.io/mosn/pkg/sync.(*workerPool).spawnWorker(0xc00064b410, 0xc000124060)
	/Users/xargin/go/src/mosn.io/mosn/pkg/sync/workerpool.go:93 +0x52
created by mosn.io/mosn/pkg/sync.(*workerPool).ScheduleAuto
	/Users/xargin/go/src/mosn.io/mosn/pkg/sync/workerpool.go:73 +0x23b
2021-01-14 18:11:48,658 [INFO] [2,c0a801041610619098656100194832] [proxy] [downstream] processError=upstreamReset, proxyId: 1, reason: UpstreamGlobalTimeout
2021-01-14 18:11:48,658 [ERROR] [2,c0a801041610619098656100194832] [stream] [http] client stream connection wait response error: EOF

2021-01-14 18:11:48,659 [INFO] [2,c0a801041610619098656100194832] [proxy] [downstream] onUpstreamReset, send hijack, reason UpstreamGlobalTimeout
2021-01-14 18:11:48,659 [WARN] [2,c0a801041610619098656100194832] [proxy] [downstream] set hijack reply, proxyId = 1, code = 504, with headers = false
goroutine 98 [running]:
runtime/debug.Stack(0xc000224780, 0x1f8, 0x3)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
mosn.io/mosn/pkg/proxy.(*downStream).handleUpstreamStatusCode(0xc000224580)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:1285 +0x80
mosn.io/mosn/pkg/proxy.(*downStream).onUpstreamHeaders(0xc000224580, 0x1)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:1264 +0x6e
mosn.io/mosn/pkg/proxy.(*upstreamRequest).receiveHeaders(0xc0002246f8, 0x1)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/upstream.go:132 +0x67
mosn.io/mosn/pkg/proxy.(*downStream).receive(0xc000224580, 0x5698ba0, 0xc000388540, 0xc000000001, 0xd, 0xc)
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:639 +0xf2b
mosn.io/mosn/pkg/proxy.(*downStream).OnReceive.func1()
	/Users/xargin/go/src/mosn.io/mosn/pkg/proxy/downstream.go:406 +0xca
mosn.io/mosn/pkg/sync.(*workerPool).spawnWorker(0xc00064b410, 0xc000124060)
	/Users/xargin/go/src/mosn.io/mosn/pkg/sync/workerpool.go:93 +0x52
created by mosn.io/mosn/pkg/sync.(*workerPool).ScheduleAuto
	/Users/xargin/go/src/mosn.io/mosn/pkg/sync/workerpool.go:73 +0x23b
```

--- a/pkg/proxy/downstream.go
+++ b/pkg/proxy/downstream.go
@@ -1219,11 +1219,6 @@ func (s *downStream) onUpstreamReset(reason types.StreamResetReason) {
 		s.requestInfo.SetResponseFlag(reasonFlag)
 		code = types.ConvertReasonToCode(reason)
 
-		if s.upstreamRequest != nil && s.upstreamRequest.host != nil {
-			s.upstreamRequest.host.HostStats().UpstreamResponseFailed.Inc(1)
-			s.upstreamRequest.host.ClusterInfo().Stats().UpstreamResponseFailed.Inc(1)
-		}
-
 		// clear reset flag
 		if log.Proxy.GetLogLevel() >= log.INFO {
 			log.Proxy.Infof(s.context, "[proxy] [downstream] onUpstreamReset, send hijack, reason %v", reason)

